src/utils/bridge/eth/history.ts

```diff
@@ -0,0 +1,351 @@
+import flatten from 'lodash/fp/flatten';
+import first from 'lodash/fp/first';
+import last from 'lodash/fp/last';
+import { ethers } from 'ethers';
+import { BridgeTxStatus, Operation } from '@sora-substrate/util';
+import {
+  api,
+  historyElementsFilter,
+  SubqueryExplorerService,
+  SUBQUERY_TYPES,
+  WALLET_CONSTS,
+} from '@soramitsu/soraneo-wallet-web';
+
+import ethersUtil from '@/utils/ethers-util';
+import { getEvmTransactionRecieptByHash, isOutgoingTransaction } from '@/utils/bridge/common/utils';
+
+import { ethBridgeApi } from '@/utils/bridge/eth/api';
+
+import { ZeroStringValue } from '@/consts';
+
+import type { Asset } from '@sora-substrate/util/build/assets/types';
+import type { BridgeHistory, NetworkFeesObject } from '@sora-substrate/util';
+import type { EthBridgeContractsAddresses } from '@/store/web3/types';
+
+const BRIDGE_ABI = new ethers.utils.Interface([
+  'function mintTokensByPeers(address tokenAddress,uint256 amount,address beneficiary,bytes32 txHash,uint8[] v,bytes32[] r,bytes32[] s,address from)',
+  'function receiveByEthereumAssetAddress(address tokenAddress,uint256 amount,address to,address from,bytes32 txHash,uint8[] v,bytes32[] r,bytes32[] s)',
+  'function receiveBySidechainAssetId(bytes32 sidechainAssetId,uint256 amount,address to,address from,bytes32 txHash,uint8[] v,bytes32[] r,bytes32[] s)',
+]);
+
+const { ETH_BRIDGE_STATES } = WALLET_CONSTS;
+
+const isLocalHistoryItem = (item: BridgeHistory, txId: string, isOutgoing: boolean, requestHash: string) => {
+  if (item.txId === txId) return true;
+
+  return isOutgoing ? item.hash === requestHash : item.externalHash === requestHash;
+};
+
+const getType = (module: string) => {
+  return module === SUBQUERY_TYPES.ModuleNames.BridgeMultisig
+    ? Operation.EthBridgeIncoming
+    : Operation.EthBridgeOutgoing;
+};
+
+// [WARNING]: api.query.ethBridge storage usage
+const getSoraHash = async (isOutgoing: boolean, requestHash: string) => {
+  return isOutgoing ? requestHash : await ethBridgeApi.getSoraHashByEthereumHash(requestHash);
+};
+
+const getSoraNetworkFee = (isOutgoing: boolean, networkFees: NetworkFeesObject) => {
+  return isOutgoing ? networkFees[Operation.EthBridgeOutgoing] : ZeroStringValue;
+};
+
+// [WARNING]: api.query.ethBridge storage usage
+const isSoraPartCompleted = async (isOutgoing: boolean, soraHash: string) => {
+  if (!isOutgoing) return true;
+
+  if (soraHash) {
+    const requestStatus = await ethBridgeApi.getRequestStatus(soraHash);
+
+    return requestStatus === BridgeTxStatus.Ready;
+  }
+
+  return false;
+};
+
+const getTransactionState = (isOutgoing: boolean, soraPartCompleted: boolean, externalHash: string, hash: string) => {
+  return isOutgoing
+    ? externalHash
+      ? ETH_BRIDGE_STATES.EVM_COMMITED
```

Comment:
I guess SONAR is right :D 
<img width="710" alt="image" src="https://github.com/sora-xor/polkaswap-exchange-web/assets/17688105/2f38e8ef-bfb8-4526-a13c-8b29d263b9d4">